code/datums/hud.dm

```diff
@@ -77,7 +79,7 @@ GLOBAL_LIST_INIT(huds, list(
 		for(var/datum/gang/G in SSticker.mode.gangs)
 			gang_huds += G.ganghud
 
-	for(var/datum/atom_hud/hud in (GLOB.huds|gang_huds))
+	for(var/datum/atom_hud/hud in (GLOB.huds|gang_huds|GLOB.active_alternate_appearances))
 		if(src in hud.hudusers)

```

Comment:
if it's possible for ```hudusers``` to be associative (that is to say, it's not already using assoc keys for some other purpose) now is a time to use that, ```in``` is way too slow.